### ISSUE TITLE
LDAP unclear

### ISSUE THREAD

**Issue by coatmaker618** [friendica/friendica#10468]
This is a pretty broad issue, I know.  So I'll try to make it a bit more specific.

I have a self-hosted Friendica instance that I would like to connect to my LDAP so I have a single sign-in for all my self-hosted web services.

The LDAP addon is installed & has a list of things that need to be added to the addon config file.  However there is no sanity verification to see if you can connect to the database, no test to see if you can see the correct users, no validation that your user has been added via autocreate or not.  And on top of this, I have no idea how to check any of these steps beyond "LDAP works on other services".

While nobody wants me to code up these tests, I'd be happy to add a page to the wiki with some more thorough instructions on setting this up and debugging once I figure it out myself :)

Thanks for the help in advance!

**Comment by MrPetovan** [comment]
The addon outputs several messages to the Friendica log. You can enable the Friendica log in the Admin settings, at a location of your choice relative to the root Friendica folder.

> While nobody wants me to code up these tests

Why not?

**Comment by coatmaker618** [comment]
> The addon outputs several messages to the Friendica log. You can enable the Friendica log in the Admin settings, at a location of your choice relative to the root Friendica folder.

*facepalm*  I didn't even think to check that.  Thanks :)

> > While nobody wants me to code up these tests
> 
> Why not?

Because whenever I try to write high level code, it ends up being as secure as the Titanic attached to the Hindenburg & I'll somehow manage to upload your passwords to facebook or something :p   

I also know almost nothing about LDAP & php >_<

**Comment by coatmaker618** [comment]
I have a user awaiting "permanent deletion" can I re-add this user now (I've deleted from userd as well)  or do I need to wait 4 days to try making an account again?

**Comment by MrPetovan** [comment]
> Because whenever I try to write high level code, it ends up being as secure as the Titanic attached to the Hindenburg & I'll somehow manage to upload your passwords to facebook or something :p

Well, we all make mistakes and we learn in the process, it shouldn't stop you from write code, especially not tests.

> I also know almost nothing about LDAP & php >_<

This, on the other hand, is a very good reason. Venturing into either of these topic is long enough, but both at the same time sounds like a small ordeal.

> I have a user awaiting "permanent deletion" can I re-add this user now (I've deleted from userd as well) or do I need to wait 4 days to try making an account again?

I'm not sure, has your node been federating with any other server? Reusing the same username of a deleted account will prevent federation with any remote server that received the deletion message. This is why the `userd` table exists in the first place.

**Comment by coatmaker618** [comment]
> I'm not sure, has your node been federating with any other server? Reusing the same username of a deleted account will prevent federation with any remote server that received the deletion message. This is why the `userd` table exists in the first place.

It has not, I've only been using this node to figure out Friendica (both as a user and an admin).  I haven't federated anywhere and only have a couple of users who are helping me test the setup.  I wanted to get LDAP working (as well as get comfortable with Friendica interface & usage & stuff) before I started federating to the real world!

> This, on the other hand, is a very good reason. Venturing into either of these topic is long enough, but both at the same time sounds like a small ordeal.

Agreed! But I appreciate the inspiration you offered as well :)

**Comment by coatmaker618** [comment]
ok, tried using the existing username that is awaiting permanent deletion and Friendica still says the name is in use.  Looks like it moves it to userd WHILE ALSO keeping it in the username table.

At least his means I don't have a cool-off period between tries!

I'll take a look at the logs and see if I see anything :)

**Comment by MrPetovan** [comment]
Without any federation, you can indeed clear the `userd` table and reset the flags of your user row after the user is deleted.

**Comment by coatmaker618** [comment]
Yup yup, that's how I've been testing!!  Just gotta figure out how to import LDAP users correctly!

**Comment by MrPetovan** [comment]
Despite not having access to an LDAP server, we've been able to refine this addon to troubleshoot a handful of other user issues, so feel free to forward any unexpected error.

**Comment by coatmaker618** [comment]
Ok, if/when I can find some more detailed error messages, I'll post them here.

Of course, please feel free to post any tests that come to mind!

**Comment by coatmaker618** [comment]
Ok, just took a look at the log when I created an account.  I don't see anything in the logs about the account at all.  Could it be that the `addon.config.php` file is not being read by the plugin? Or is it more likely that there's something else?

Also, I do have autocreate accounts from LDAP on.  So maybe that's conflicting? My understanding is that in theory that should mean I don't even need to create an account?
